homeassistant/components/template/lawn_mower.py

```diff
@@ -0,0 +1,204 @@
+        }
+    )
+    .extend(TEMPLATE_ENTITY_ATTRIBUTES_SCHEMA_LEGACY.schema)
+    .extend(TEMPLATE_ENTITY_AVAILABILITY_SCHEMA_LEGACY.schema),
+)
+
+PLATFORM_SCHEMA = cv.PLATFORM_SCHEMA.extend(
+    {vol.Required(CONF_LAWN_MOWERS): vol.Schema({cv.slug: LAWN_MOWER_SCHEMA})}
+)
+
+
+async def _async_create_entities(hass: HomeAssistant, config: ConfigType):
+    """Create the Lawn Mowers."""
+    lawn_mowers = []
+
+    for object_id, entity_config in config[CONF_LAWN_MOWERS].items():
+        entity_config = rewrite_common_legacy_to_modern_conf(entity_config)
+        unique_id = entity_config.get(CONF_UNIQUE_ID)
+
+        lawn_mowers.append(
+            TemplateLawnMower(
+                hass,
+                object_id,
+                entity_config,
+                unique_id,
+            )
+        )
+
+    return lawn_mowers
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the template lawn mowers."""
+    async_add_entities(await _async_create_entities(hass, config))
+
+
+class TemplateLawnMower(TemplateEntity, LawnMowerEntity):
+    """A template lawn mower component."""
+
+    _attr_should_poll = False
+
+    def __init__(
+        self,
+        hass,
+        object_id,
+        config,
+        unique_id,
+    ):
+        """Initialize the lawn mower."""
+        super().__init__(
+            hass, config=config, fallback_name=object_id, unique_id=unique_id
+        )
+        self.entity_id = async_generate_entity_id(
+            ENTITY_ID_FORMAT, object_id, hass=hass
+        )
+        friendly_name = self._attr_name
+
+        self._template = config.get(CONF_VALUE_TEMPLATE)
+
+        self._start_mowing_script = None
+        if start_mowing_action := config.get(SERVICE_START_MOWING):
+            self._start_mowing_script = Script(
+                hass, start_mowing_action, friendly_name, DOMAIN
+            )
+            self._attr_supported_features |= LawnMowerEntityFeature.START_MOWING
+
+        self._pause_script = None
+        if pause_action := config.get(SERVICE_PAUSE):
+            self._pause_script = Script(hass, pause_action, friendly_name, DOMAIN)
+            self._attr_supported_features |= LawnMowerEntityFeature.PAUSE
+
+        self._dock_script = None
+        if dock_action := config.get(SERVICE_DOCK):
+            self._dock_script = Script(hass, dock_action, friendly_name, DOMAIN)
+            self._attr_supported_features |= LawnMowerEntityFeature.DOCK
+
+        self._attr_activity = None
+
+    @property
+    def activity(self) -> LawnMowerActivity | None:
+        """Return the status of the lawn mower."""
+        return self._attr_activity
+
+    async def async_start_mowing(self) -> None:
+        """Start or resume the mowing task."""
+        if self._start_mowing_script is None:
+            return
```

Comment:
I don't think this needs to be checked, it should not be possible to call the service if the `LawnMowerEntityFeature.START_MOWING` feature flag is not set.

If mypy complains, we should do this instead:

```
if TYPE_CHECKING:
    # assert self._start_mowing_script is not None
```

Same for the other scripts